pkg/configurator/methods.go

```diff
@@ -124,17 +125,29 @@ func (c *client) GetEnvoyLogLevel() string {
 
 // GetEnvoyImage returns the envoy image
 func (c *client) GetEnvoyImage() string {
-	return c.getMeshConfig().Spec.Sidecar.EnvoyImage
+	image := c.getMeshConfig().Spec.Sidecar.EnvoyImage
+	if image == "" {
+		image = os.Getenv("OSM_ENVOY_IMAGE")
```

Comment:
I remember @shashankram mentioning when we had a default before, the default would be used if there was some error in `getMeshConfig()` since it returns an empty MeshConfig then. For users that override the image to pull from their own secure registries, using the default in that case probably isn't what users would expect. Since users can control what the default is here though, maybe that's less of a concern.

Reply:
I think that would probably be a documentation exercise, right? If the user specifies `osm.sidecarImage` and `osm.sidecarWindowsImage` when they installed OSM then it would effectively set the defaults to whatever they want. That way if we get no data back from the MeshConfig then it falls back to their defaults.

Reply:
Yeah, that sounds reasonable. Whether or not we should fail to inject if looking up the image names in the MeshConfig fails seems like a separate discussion.